### ISSUE TITLE
Column selection fails for nested fields and field names containing hyphen (regression at 7e43bf1)

### ISSUE THREAD

**Issue by burpheart** [brimdata/brim#2556]
Before opening a new issue, please make sure you've reviewed the troubleshooting guide:
https://github.com/brimdata/brim/wiki/Troubleshooting

**Describe the bug**

When a column name has special characters such as "-" or is part of a complex structured nesting, the column cannot be selected or unselected by the column selector.
**To Reproduce**

```
{"content-length":275,"test":{"abc":"123","d":{"c":"456"}}}
{"content-length":796,"test":{"abc":"123","d":{"c":"456"}}}
{"content-length":275,"test":{"abc":"123","d":{"c":"456"}}}
```
The column "content-length"  "test › abc" "test › d › c" cannot be selected or unselected by the column selector.

**Expected behavior**
Tt can be selected as a column in column selector.

**Screenshots**

**Desktop Info**
[Please complete the following information]
 - OS: Windows
 - Brim Version: 0.31.0

**Comment by philrz** [comment]
I've confirmed that this issue is not unique to JSON data and is a regression starting with Brim commit 7e43bf1 which is associated with #1856.

Below is some Zeek TSV test data [one-weird.zeek.gz](https://github.com/brimdata/brim/files/9659309/one-weird.zeek.gz) which can be used in an alternate repro. Fields in Zeek data typically don't contain the `-` character and that's the data we were primarily importing back when this regression occurred, so I've cooked up a fake field that contains it in addition to chopping off the rest of the record.

```
$ gzcat one-weird.zeek 
#separator \x09
#set_separator	,
#empty_field	(empty)
#unset_field	-
#path	weird
#fields	ts	id.orig_h	has-slash
#types	string	string	int
2018-03-24T17:15:20.600843Z	192.168.0.1	1
```

Here's a repro with current Brim commit 6eaaa54 where we see that I can uncheck/check the fields like `_path` and `ts` but not the nested `id.orig_h` nor the hypen-containing `has-slash`.

https://user-images.githubusercontent.com/5934157/192626546-1f1e3cf6-e2cb-41c5-98be-73f42e1e56a4.mp4

And here's the evidence of when it was last working at commit 564f8ac which was right before the changes merged for #1856.

https://user-images.githubusercontent.com/5934157/192627025-711714f8-9346-4140-afa4-d1b046194fa9.mp4

**Comment by philrz** [comment]
This one is on track to be fixed by the changes in #2626. As shown in the attached video at commit ee252c5 from that branch, now all the columns can be unselected/selected regardless of whether they're nested fields or contain hyphens.

https://user-images.githubusercontent.com/5934157/214711953-1ec4903a-23e2-445a-a6dd-f55c866b317a.mp4

**Comment by philrz** [comment]
Verified in Brim commit 499a2c9.

As shown in the attached video that uses the original repro data, in the new design the "eye" control in the column chooser works for the nested field and the hyphen-containing field.

https://user-images.githubusercontent.com/5934157/214936912-1ed33f2e-a71a-4ea6-9430-fbd63476da59.mp4

Thanks @jameskerr!
